.gitlab/includes/clang11_pipeline.yml

```
@@ -20,26 +20,45 @@ include:
                   -DPIKA_WITH_MAX_CPU_COUNT=256 -DPIKA_WITH_MALLOC=system \
                   -DPIKA_WITH_SPINLOCK_DEADLOCK_DETECTION=ON"
 
-clang11_spack_image:
-  stage: spack_configs
+clang11_spack_compiler_image:
+  stage: spack_compiler
   needs: [base_spack_image]
   extends:
     - .container-builder
     - .variables_clang11_config
   before_script:
     - export DOCKERFILE_SHA=`sha256sum $DOCKERFILE | head -c 16`
-    - CONFIG_TAG=`echo $DOCKERFILE_SHA-$ARCH-$BASE_IMAGE-$COMPILER-$SPACK_SPEC | sha256sum - | head -c 16`
+    - CONFIG_TAG=`echo $DOCKERFILE_SHA-$BASE_IMAGE-$SPACK_SPEC | sha256sum - | head -c 16`
     - compiler=${COMPILER/@/-}
     - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/base/pika-$compiler:$CONFIG_TAG
     - echo -e "compiler=$compiler" >> compiler.env
     - echo -e "BASE_IMAGE=$PERSIST_IMAGE_NAME" >> compiler.env
   variables:
     DOCKERFILE: .gitlab/docker/Dockerfile.spack_compiler
-    DOCKER_BUILD_ARGS: '["BASE_IMAGE","ARCH","COMPILER","SPACK_SPEC"]'
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE","ARCH","COMPILER"]'
   artifacts:
     reports:
       dotenv: compiler.env
 
+clang11_spack_image:
+  stage: spack_dependencies
+  needs: [clang11_spack_compiler_image]
+  extends:
+    - .container-builder
+    - .variables_clang11_config
+  before_script:
+    - export DOCKERFILE_SHA=`sha256sum $DOCKERFILE | head -c 16`
+    - CONFIG_TAG=`echo $DOCKERFILE_SHA-$ARCH-$BASE_IMAGE-$COMPILER-$SPACK_SPEC | sha256sum - | head -c 16`
+    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/base/pika-$compiler-dependencies:$CONFIG_TAG
+    - echo -e "compiler=$compiler" >> dependencies.env
+    - echo -e "BASE_IMAGE=$PERSIST_IMAGE_NAME" >> dependencies.env
+  variables:
+    DOCKERFILE: .gitlab/docker/Dockerfile.spack_dependencies
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE","SPACK_SPEC"]'
```

This is backwards and should be:
```suggestion
    - CONFIG_TAG=`echo $DOCKERFILE_SHA-$BASE_IMAGE-$SPACK_SPEC | sha256sum - | head -c 16`
    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/base/pika-$compiler-dependencies:$CONFIG_TAG
    - echo -e "compiler=$compiler" >> dependencies.env
    - echo -e "BASE_IMAGE=$PERSIST_IMAGE_NAME" >> dependencies.env
  variables:
    DOCKERFILE: .gitlab/docker/Dockerfile.spack_dependencies
    DOCKER_BUILD_ARGS: '["BASE_IMAGE","SPACK_SPEC"]'
```
since `SPACK_SPEC` and `BASE_IMAGE` are the only `ARG`s in this stage.